lib/Basic/PlatformUtility.cpp

```diff
@@ -42,11 +44,68 @@ int sys::close(int fileHandle) {
 #endif
 }
 
+#if defined(_WIN32)
+time_t filetimeToTime_t(FILETIME ft) {
+  long long ltime = ft.dwLowDateTime | (ft.dwHighDateTime << 32);
+  return (time_t)((ltime - 116444736000000000) / 10000000);
+}
+#endif
+
 int sys::lstat(const char *fileName, sys::StatStruct *buf) {
 #if defined(_WIN32)
-  // We deliberately ignore lstat on Windows, and delegate
-  // to stat.
-  return ::_stat(fileName, buf);
+  llvm::SmallVector<UTF16, 20> wfilename;
```

Comment:
This is definitely not a _good_ implementation of lstat, but Windows doesn't provide it and it's good enough for the purposes of checking if file is a directory/regular file/symlink so llbuild knows to recurse into it or not.